src/AgentVisibilityContext.tsx

```diff
@@ -1,90 +1,71 @@
-import React, { createContext, useEffect, useMemo } from 'react';
-import { CheckboxState, VisibilitySelectionMap } from './constants';
+import React, { createContext, useState, ReactNode } from 'react';
 import { UIDisplayData } from '@aics/simularium-viewer';
 
-import { getNewHiddenAgents } from './selectors';
+import { VisibilitySelectionMap } from './constants';
+import { getNewSelectionMap, mapUIDisplayDataToSelectionMap } from './utils';
+import { SelectionType } from './types';
 
 interface VisibilityContextType {
   uiDisplayData: UIDisplayData;
   hiddenAgents: VisibilitySelectionMap;
-  allAgentsHidden: VisibilitySelectionMap;
-  noAgentsHidden: VisibilitySelectionMap;
+  highlightedAgents: VisibilitySelectionMap;
   receiveUIDisplayData: (data: UIDisplayData) => void;
-  handleAllAgentsCheckboxChange: (hiddenState: CheckboxState) => void;
-  handleAgentCheckboxChange: (agentName: string) => void;
+  setHiddenAgents: React.Dispatch<React.SetStateAction<VisibilitySelectionMap>>;
+  toggleAgentVisibility: (
+    agentName: string,
+    selectionType: SelectionType
+  ) => void;
 }
 
 export const VisibilityContext = createContext<VisibilityContextType>({
   uiDisplayData: [],
   hiddenAgents: {},
-  allAgentsHidden: {},
-  noAgentsHidden: {},
+  highlightedAgents: {},
   receiveUIDisplayData: () => {},
-  handleAllAgentsCheckboxChange: () => {},
-  handleAgentCheckboxChange: () => {},
+  setHiddenAgents: () => {},
+  toggleAgentVisibility: () => {},
```

Comment:
just reading these two alone I have a hard time figuring out how they're different and why they're both needed. They sound like synonyms